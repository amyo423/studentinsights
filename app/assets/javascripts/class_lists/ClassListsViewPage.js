import React from 'react';
import Timestamp from '../components/Timestamp';
import School from '../components/School';
import Educator from '../components/Educator';
import GenericLoader from '../components/GenericLoader';
import SectionHeading from '../components/SectionHeading';
import {gradeText} from '../helpers/gradeText';
import {bucketInto} from '../helpers/bucketInto';
import {fetchAllClassListsJson} from './api';


// Show users their class lists.  More useful for principals, building admin,
// or ELL/SPED teachers than classroom teachers (who are typically
// making a single list).
export default class ClassListsViewPage extends React.Component {
  constructor(props) {
    super(props);

    this.renderPage = this.renderPage.bind(this);
  }

  render() {
    return (
      <div className="ClassListsViewPage">
        <SectionHeading>Class lists</SectionHeading>
        <GenericLoader
          promiseFn={fetchAllClassListsJson}
          render={this.renderPage} />
      </div>
    );
  }

  renderPage(json) {
    const classLists = json.class_lists;

    if (classLists.length === 0) return <div>None!</div>;

    return (
      <table>
        <thead>
          <tr>
            <th>School</th>
            <th>Grade next year</th>
            <th>Owner</th>
            <th>Created on</th>
            <th>Session</th>
            <th>{'\u00A0'}</th>
          </tr>
        </thead>
        <tbody>{classLists.map(classList => {
          const shorthand = bucketInto(classList.workspace_id, SHORTHANDS);
          return (
            <tr key={classList.id}>
              <td><School {...classList.school} /></td>
              <td>{gradeText(classList.grade_level_next_year)}</td>
              <td><Educator educator={classList.created_by_educator} /></td>
              <td><Timestamp railsTimestamp={classList.created_at} /></td>
              <td><a href={`/classlists/${classList.id}`}>view {classList.workspace_id.slice(0, 4).toUpperCase()}...</a></td>
            </tr>
          );
        })}</tbody>
      </table>
    );
  }
}

const SHORTHANDS = ["AliceBlue","AntiqueWhite","Aqua","Aquamarine","Azure","Beige","Bisque","Black","BlanchedAlmond","Blue","BlueViolet","Brown","BurlyWood","CadetBlue","Chartreuse","Chocolate","Coral","CornflowerBlue","Cornsilk","Crimson","Cyan","DarkBlue","DarkCyan","DarkGoldenRod","DarkGray","DarkGrey","DarkGreen","DarkKhaki","DarkMagenta","DarkOliveGreen","Darkorange","DarkOrchid","DarkRed","DarkSalmon","DarkSeaGreen","DarkSlateBlue","DarkSlateGray","DarkSlateGrey","DarkTurquoise","DarkViolet","DeepPink","DeepSkyBlue","DimGray","DimGrey","DodgerBlue","FireBrick","FloralWhite","ForestGreen","Fuchsia","Gainsboro","GhostWhite","Gold","GoldenRod","Gray","Grey","Green","GreenYellow","HoneyDew","HotPink","IndianRed","Indigo","Ivory","Khaki","Lavender","LavenderBlush","LawnGreen","LemonChiffon","LightBlue","LightCoral","LightCyan","LightGoldenRodYellow","LightGray","LightGrey","LightGreen","LightPink","LightSalmon","LightSeaGreen","LightSkyBlue","LightSlateGray","LightSlateGrey","LightSteelBlue","LightYellow","Lime","LimeGreen","Linen","Magenta","Maroon","MediumAquaMarine","MediumBlue","MediumOrchid","MediumPurple","MediumSeaGreen","MediumSlateBlue","MediumSpringGreen","MediumTurquoise","MediumVioletRed","MidnightBlue","MintCream","MistyRose","Moccasin","NavajoWhite","Navy","OldLace","Olive","OliveDrab","Orange","OrangeRed","Orchid","PaleGoldenRod","PaleGreen","PaleTurquoise","PaleVioletRed","PapayaWhip","PeachPuff","Peru","Pink","Plum","PowderBlue","Purple","Red","RosyBrown","RoyalBlue","SaddleBrown","Salmon","SandyBrown","SeaGreen","SeaShell","Sienna","Silver","SkyBlue","SlateBlue","SlateGray","SlateGrey","Snow","SpringGreen","SteelBlue","Tan","Teal","Thistle","Tomato","Turquoise","Violet","Wheat","White","WhiteSmoke","Yellow","YellowGreen"];

const styles = {
  
};
