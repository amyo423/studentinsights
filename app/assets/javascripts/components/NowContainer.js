import React from 'react';

// A container for testing that allows passing in a `nowFn`
// that will be shared with child components through context.
export default class NowContainer extends React.Component {
  getChildContext() {
    const {nowFn} = this.props;
    return {
      nowFn() { return nowFn(); }
    };
  }

  render() {
    const {children} = this.props;

    return children;
  }
}
NowContainer.childContextTypes = {
  nowFn: React.PropTypes.func
};
NowContainer.propTypes = {
  children: React.PropTypes.node.isRequired,
  nowFn: React.PropTypes.func.isRequired
};