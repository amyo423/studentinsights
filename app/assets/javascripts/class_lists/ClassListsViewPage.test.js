import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {withDefaultNowContext} from '../../../../spec/javascripts/support/NowContainer';
import mockWithFixtures from './fixtures/mockWithFixtures';
import ClassListsViewPage, {ClassListsViewPageView} from './ClassListsViewPage';
import workspaces_json from './fixtures/workspaces_json';

beforeEach(() => mockWithFixtures());

export function testProps(props = {}) {
  return {
    currentEducatorId: 999999,
    ...props
  };
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  const props = testProps();
  ReactDOM.render(withDefaultNowContext(<ClassListsViewPage {...props} />), el);
});


it('snapshots view', () => {
  const props = testProps({...workspaces_json});
  const tree = renderer
    .create(withDefaultNowContext(<ClassListsViewPageView {...props} />))
    .toJSON();
  expect(tree).toMatchSnapshot();
});