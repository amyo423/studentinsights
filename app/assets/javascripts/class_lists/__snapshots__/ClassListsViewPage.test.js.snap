// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots view 1`] = `
<div>
  <div
    className="SectionHeading"
    style={
      Object {
        "borderBottom": "1px solid #333",
        "padding": 10,
      }
    }
  >
    <h4
      style={
        Object {
          "color": "black",
          "display": "inline",
        }
      }
    >
      Class lists
    </h4>
  </div>
  <div>
    <div
      className="Hover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={undefined}
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#4A90E2",
            "border": "none",
            "borderRadius": 28,
            "color": "white",
            "cursor": "pointer",
            "display": "block",
            "fontSize": 14,
            "margin": 10,
            "padding": "8px 25px",
            "textDecoration": "none",
          }
        }
      >
        New list
      </button>
    </div>
    <table
      style={
        Object {
          "border": "1px solid #eee",
          "borderCollapse": "collapse",
          "margin": 10,
        }
      }
    >
      <thead>
        <tr>
          <th
            style={
              Object {
                "backgroundColor": "#ccc",
                "border": "1px solid #eee",
                "fontWeight": "bold",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            School
          </th>
          <th
            style={
              Object {
                "backgroundColor": "#ccc",
                "border": "1px solid #eee",
                "fontWeight": "bold",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Grade next year
          </th>
          <th
            style={
              Object {
                "backgroundColor": "#ccc",
                "border": "1px solid #eee",
                "fontWeight": "bold",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Owner
          </th>
          <th
            style={
              Object {
                "backgroundColor": "#ccc",
                "border": "1px solid #eee",
                "fontWeight": "bold",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Created on
          </th>
          <th
            style={
              Object {
                "backgroundColor": "#ccc",
                "border": "1px solid #eee",
                "fontWeight": "bold",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Revisions
          </th>
          <th
            style={
              Object {
                "backgroundColor": "#ccc",
                "border": "1px solid #eee",
                "fontWeight": "bold",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Workspace
          </th>
          <th
            style={
              Object {
                "backgroundColor": "#ccc",
                "border": "1px solid #eee",
                "fontWeight": "bold",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="School"
              href="/schools/2"
              style={Object {}}
            >
              Arthur D Healey
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            3rd grade
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="Educator"
              href="mailto:uri@demo.studentinsights.org"
              style={Object {}}
            >
              Uri Disney
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Wednesday 5/9
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            17
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <pre>
              6305
              -
              345
            </pre>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              href="/classlists/63055567-a2e1-49b3-a6da-655658445177"
            >
              open
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="School"
              href="/schools/2"
              style={Object {}}
            >
              Arthur D Healey
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            5th grade
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="Educator"
              href="mailto:uri@demo.studentinsights.org"
              style={Object {}}
            >
              Uri Disney
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Tuesday 5/8
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            9
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <pre>
              fda9
              -
              328
            </pre>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              href="/classlists/fda99ce2-99be-4f39-a56e-86c16a790caf"
            >
              open
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="School"
              href="/schools/2"
              style={Object {}}
            >
              Arthur D Healey
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            6th grade
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="Educator"
              href="mailto:sarah@demo.studentinsights.org"
              style={Object {}}
            >
              Sarah Teacher
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Tuesday 5/8
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            26
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <pre>
              06d3
              -
              319
            </pre>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              href="/classlists/06d32485-77a5-4724-8780-d55df3aedebf"
            >
              open
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="School"
              href="/schools/2"
              style={Object {}}
            >
              Arthur D Healey
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            6th grade
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="Educator"
              href="mailto:uri@demo.studentinsights.org"
              style={Object {}}
            >
              Uri Disney
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Tuesday 5/8
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            4
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <pre>
              8893
              -
              293
            </pre>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              href="/classlists/8893e4a1-b57c-47e6-89b6-1d767aef5d21"
            >
              open
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="School"
              href="/schools/2"
              style={Object {}}
            >
              Arthur D Healey
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            3rd grade
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="Educator"
              href="mailto:uri@demo.studentinsights.org"
              style={Object {}}
            >
              Uri Disney
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Tuesday 5/8
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            7
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <pre>
              191c
              -
              289
            </pre>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              href="/classlists/191c6264-0352-453e-a495-481e93e19c80"
            >
              open
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="School"
              href="/schools/2"
              style={Object {}}
            >
              Arthur D Healey
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            6th grade
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="Educator"
              href="mailto:sarah@demo.studentinsights.org"
              style={Object {}}
            >
              Sarah Teacher
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Monday 5/7
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            3
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <pre>
              a0f1
              -
              282
            </pre>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              href="/classlists/a0f1dcb2-96f6-4b6f-bbf7-a21a0fa2225c"
            >
              open
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="School"
              href="/schools/2"
              style={Object {}}
            >
              Arthur D Healey
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            3rd grade
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="Educator"
              href="mailto:uri@demo.studentinsights.org"
              style={Object {}}
            >
              Uri Disney
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Monday 5/7
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            7
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <pre>
              1ffe
              -
              279
            </pre>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              href="/classlists/1ffe068d-6c62-4e1f-81b6-180b334f6d93"
            >
              open
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="School"
              href="/schools/2"
              style={Object {}}
            >
              Arthur D Healey
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            3rd grade
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="Educator"
              href="mailto:uri@demo.studentinsights.org"
              style={Object {}}
            >
              Uri Disney
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Monday 5/7
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            6
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <pre>
              9115
              -
              272
            </pre>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              href="/classlists/9115a670-b0dc-4b31-bdfd-64ef73bf34e1"
            >
              open
            </a>
          </td>
        </tr>
        <tr>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="School"
              href="/schools/2"
              style={Object {}}
            >
              Arthur D Healey
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            3rd grade
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              className="Educator"
              href="mailto:uri@demo.studentinsights.org"
              style={Object {}}
            >
              Uri Disney
            </a>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            Monday 5/7
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            26
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <pre>
              294f
              -
              266
            </pre>
          </td>
          <td
            style={
              Object {
                "border": "1px solid #eee",
                "padding": 10,
                "textAlign": "left",
                "verticalAlign": "top",
              }
            }
          >
            <a
              href="/classlists/294f6a1e-42db-4fbb-b283-0acd6c328f4b"
            >
              open
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
